From sarah.a.sharp@linux.intel.com  Fri May  1 14:59:43 2009
From: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Date: Wed, 29 Apr 2009 19:04:59 -0700
Subject: USB: xhci: Fix Link TRB handoff bit twiddling.
To: Greg KH <greg@kroah.com>
Message-ID: <20090430020459.GA31186@gamba.jf.intel.com>
Content-Disposition: inline


Make sure to preserve all bits *except* the TRB_CHAIN bit when giving a
Link TRB to the hardware.  We need to save things like TRB type and the
toggle bit in the control dword.

Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/xhci-ring.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -188,7 +188,7 @@ static void inc_enq(struct xhci_hcd *xhc
 					next->link.control &= (u32) ~TRB_CYCLE;
 				else
 					next->link.control |= (u32) TRB_CYCLE;
-				next->link.control &= TRB_CHAIN;
+				next->link.control &= ~TRB_CHAIN;
 				next->link.control |= chain;
 			}
 			/* Toggle the cycle bit after the last ring segment. */
